src/api/cvc4cpp.h

```diff
@@ -1104,29 +1105,38 @@ class CVC4_PUBLIC DatatypeDecl
  public:
   /**
    * Constructor.
+   * @param s the solver that created this datatype declaration
    * @param name the name of the datatype
    * @param isCoDatatype true if a codatatype is to be constructed
    * @return the DatatypeDecl
    */
-  DatatypeDecl(const std::string& name, bool isCoDatatype = false);
+  DatatypeDecl(const Solver* s,
```

Comment:
These constructors should be private, users should use the `mk*` methods of the solver.